README.md

```diff
@@ -1,277 +1,899 @@
+Many DEXs on Cardano are currently implemented in ways that lock users' assets into a tightly fixed,
+and/or centrally maintained, set of script addresses and UTxOs. This *concentrated dApp* design
+pattern is reminiscent of Ethereum's accounts-based programming paradigm.
+
+However, the concentrated dApp design does not map well to eUTxO based blockchains. To be exact, a
+major consequence of pooling assets into a predefined set of UTxOs, which are typically referred to
+as liquidity pools (LPs), is that the DEX has a significant concurrency bottleneck since all users
+must share the one-time-use UTxOs. The ultimate result of this bottleneck is that users are not
+allowed to directly interact with the DEX. Instead, users are required to go through (usually
+permissioned) middle-men called batchers. 
+
+As an alternative to LP based DEXs, order-book style DEXs give every user their own UTxO with their
+own terms. This approach lends itself much better to eUTxO based blockchains. However, this design
+currently suffers from similar availability challenges due to the difficulting of finding and
+matching open orders. In order to workaround this limitation, order-book style DEXs also tend to
+resort to the concentrated dApp design since it is easier to find orders if they are located at a
+single address (or a few addresses) than if they were scattered across many addresses. Furthermore,
+and more importantly, current order-book style DEXs suffer from a lack of liquidity due to not
+having a way to incentive users to provide liquidity to the DEX. This lack of early liquidity is why
+most DEXs currently use the LP based DEX architecture. 
+
+This result is unfortunate since LP based DEXs are entirely unsuitable to be used as the foundation
+of a trustless and decentralized economy, especially those built on Proof-of-Stake (PoS)
+blockchains.
+
+##### Why the LP based architecture is not the right architecture for DEXs (in no particular order)
+
+- No matter how performant a system of batchers is, their resources do *not* scale in proportion to
+the number of users unless new batchers can permissionlessly join when demand is high. 
+- Since an economy's health is proportional to how easily and accurately price discovery can occur,
+DEXs that do not allow users to freely express their own desired prices can only result in an
+unhealthy blockchain economy. Furthermore, impermanent loss (a consequence of not being able to
+express desired prices) makes it impossible for users to properly manage risk.
+- Batcher based protocols lack trustless composability with other dApps; a critical feature for both
+scaling eUTxO-based DeFi and building a fully featured, trustless blockchain economy.
+- The concentration of assets into a single address or few addresses directly undermines the
+security assumptions of PoS blockchains. This ultimately means it is *impossible* for LPs to serve
+as the foundation of a trustless and decentralized economy.
+
```

Comment:
Perhaps add:
The LP based architecture pools user assets at an address outside of their direct control.  This forces users to choose between using their assets to provide trading liquidity or being able to harness the voting power their assets provide.

Reply:
What if I added this bullet to the end:

> - The concentration of assets forces uses to give up voting rights over their assets. In essence, users are forced to pick between participating in DeFi and participating in decentralized governance. Furthermore, whichever entity controls the voting power of the concentrated assets has tremendous influence over the decentralized government.

Reply:
I think this point is so important because it not only limits user voting power, but it also fundamentally undermines Cardano's security.  I like what you have said already, but I somehow want it to be harder hitting.

Reply:
What about:

> - The concentration of assets into a single address or few addresses directly undermines the
security assumptions of PoS blockchains.
> - The concentration of assets forces users to give up voting rights over their assets. In essence, users are forced to pick between participating in DeFi and participating in decentralized governance. Furthermore, whichever entity controls the voting power of the concentrated assets has tremendous influence over the decentralized government.
> 
> The last two bullets especially mean it is *impossible* for LPs to serve as the foundation of a trustless and decentralized economy.

I think the final sentence is true and emphasizes it well. I'm not sure how to emphasize it better than that.

Reply:
The current DEX design is quite simply fundamentally flawed for a POS blockchain.  Consequently the DEX owners come up with these "governance" tokens and "staking decision" tokens.

Cardano-swaps design is beautiful by comparison.